finder/index_test.go

```diff
@@ -36,16 +40,26 @@ func Test_useReverseWithFixedDepth(t *testing.T) {
 		depth  int
 		result bool
 	}{
-		{"a.b.c.d.e", 0, false},
+		{"a*.b.c.d.e", -1, false},
+		{"a*.b.c.d.e", 0, true},
+		{"a.b*.c.d.e", 0, false},
+		{"a.b.c.d.e*", 0, false},
+		{"a.b*", 1, false},
+		{"a*.b", 1, true},
 		{"a.b.c.d.e", 1, false},
 		{"a.b.c.d.e*", 1, false},
-		{"a.b.c.d*.e", 1, true},
+		{"a.b.c.d*.e", 1, false},
+		{"a.b*.c.d.e", 1, true},
+		{"a.b.c.*.e.*.j", 1, false},
 		{"a.b.c.d*.e", 2, false},
 		{"a*.b.c.d*.e", 2, true}, // Wildcard at first level, use reverse if possible
 		{"a.b*.c.d*.e", 2, false},
+		{"a.b*.c.d", 2, true},
+		{"a.b.c*.d", 2, false},
+		{"a.b.c*.d.e", 2, true},
 		{"a.*.c.*.e.*.j", 2, false},
-		{"a.*.c.*.e.*.j", 1, true},
-		{"a.b*.c.*d.e", 2, false},
```

Comment:
Why remove this testcase ? It's a case for default settings. And it's not trivial for choose reverse or forward index in this case.